sdks/python/apache_beam/yaml/yaml_ml.py

```diff
@@ -33,11 +40,411 @@
+    def apply(constructor):
+      cls.handler_types[type_name] = constructor
+      return constructor
+
+    return apply
+
+  @classmethod
+  def create_handler(cls, model_handler_spec) -> "ModelHandlerProvider":
+    typ = model_handler_spec['type']
+    config = model_handler_spec['config']
+    try:
+      result = cls.handler_types[typ](**config)
+      if not hasattr(result, 'to_json'):
+        result.to_json = lambda: model_handler_spec
+      return result
+    except Exception as exn:
+      raise ValueError(
+          f'Unable to instantiate model handler of type {typ}. {exn}')
+
+
+@ModelHandlerProvider.register_handler_type('VertexAIModelHandlerJSON')
+class VertexAIModelHandlerJSONProvider(ModelHandlerProvider):
+  def __init__(
+      self,
+      endpoint_id: str,
+      project: str,
+      location: str,
+      preprocess: Dict[str, str],
+      experiment: Optional[str] = None,
+      network: Optional[str] = None,
+      private: bool = False,
+      min_batch_size: Optional[int] = None,
+      max_batch_size: Optional[int] = None,
+      max_batch_duration_secs: Optional[int] = None,
+      env_vars: Optional[Dict[str, Any]] = None,
+      postprocess: Optional[Dict[str, str]] = None):
+    """
+    ModelHandler for Vertex AI.
+
+    For example: ::
+
+        - type: RunInference
+          config:
+            inference_tag: 'my_inference'
+            model_handler:
+              type: VertexAIModelHandlerJSON
+              config:
+                endpoint_id: 9876543210
+                project: my-project
+                location: us-east1
+                preprocess:
+                  callable: 'lambda x: {"prompt": x.prompt, "max_tokens": 50}'
+
+    Args:
+      endpoint_id: the numerical ID of the Vertex AI endpoint to query.
+      project: the GCP project name where the endpoint is deployed.
+      location: the GCP location where the endpoint is deployed.
+      experiment: Experiment label to apply to the
+        queries. See
+        https://cloud.google.com/vertex-ai/docs/experiments/intro-vertex-ai-experiments
+        for more information.
+      network: The full name of the Compute Engine
+        network the endpoint is deployed on; used for private
+        endpoints. The network or subnetwork Dataflow pipeline
+        option must be set and match this network for pipeline
+        execution.
+        Ex: "projects/12345/global/networks/myVPC"
+      private: If the deployed Vertex AI endpoint is
+        private, set to true. Requires a network to be provided
+        as well.
+      min_batch_size: The minimum batch size to use when batching
+        inputs.
+      max_batch_size: The maximum batch size to use when batching
+        inputs.
+      max_batch_duration_secs: The maximum amount of time to buffer
+        a batch before emitting; used in streaming contexts.
+      env_vars: Environment variables.
+      preprocess: A python callable, defined either inline, or using a file,
```

Comment:
Nit: args don't match the order defined in the function, and specifically you have optional args before required args.

Reply:
Ah yes, that was the order before I made it required, thanks!